logging/test/test.go

```diff
@@ -39,6 +40,7 @@ func (l *Logger) WithFields(fields map[string]interface{}) logging.Logger {
 		level:   l.level,
 		entries: l.entries,
 		fields:  l.fields,
+		mtx:     l.mtx,
```

Comment:
Curious if this is why we need `mtx` to be a pointer?

Reply:
It is, yes. Previously, `WithFields()` would create a new logger that shared e.g. `entries` with the original logger. But as it'd get it's own mutex instance, which it then would use to protect access to `entries`, race conditions could occur between the "parent" and "child" loggers. Concretely, this would make it possible to not pass race checks on tests where the logger is in use concurrently to `Entries()` being called.
We might not have many older tests like that, which might have hid the issue; but e.g. `TestExecV1Compatible` will make it a much more likely issue.

This is only a change to the test logger, so any performance degradation (we are sharing the same mutex over more instances, after all) will only be visible in test execution time, and not in production.